flocker/provision/test/test_digitalocean.py

```diff
@@ -0,0 +1,117 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Tests for ``flocker.provision._digitalocean``.
+"""
+import copy
+import os
+from unittest import skipUnless
+
+try:
+    import pyocean
+except ImportError:
+    PYOCEAN_INSTALLED = False
+else:
+    PYOCEAN_INSTALLED = True
+
+from twisted.trial.unittest import SynchronousTestCase, SkipTest
+
+from flocker.provision._digitalocean import (
+    set_droplet_kernel, retry_if_pending, latest_droplet_kernel,
+    kernel_from_digitalocean_version, DIGITALOCEAN_KERNEL)
+from flocker.testtools import random_name
+
+
+TESTING_DROPLET_ATTRIBUTES = {
+    'region': 'lon1',
+    'size': '8gb',
+    'image': 'fedora-20-x64'
+}
+
+
+@skipUnless(PYOCEAN_INSTALLED, "pyocean not installed")
```

Comment:
It raises `SkipTest` which bubbles up and is caught by the test runner. This seemed like a good place for it as it's the only place that actually requires pyocean. On the other hand it would be a bad idea if the function was called after a lengthy setup routine. It's not in this case though so I think I'll leave it.